src/main.cc

```diff
@@ -14,6 +15,35 @@
 
 namespace bpo = boost::program_options;
 
+static const char* const AkliteSystemLock{"/var/lock/aklite.lock"};
+
+class FileLock {
+ public:
+  explicit FileLock(const char* path = AkliteSystemLock) : path_{path}, file_{path}, lock_{path} {
+    if (!lock_.try_lock()) {
+      throw std::runtime_error("Failed to obtain the aklite lock, another instance of aklite must be running !!!");
+    }
+  }
+
+  ~FileLock() {
+    try {
+      lock_.unlock();
+      boost::filesystem::remove(path_);
+    } catch (const std::exception& exc) {
+      LOG_WARNING << "Failed to unlock, " << path_ << ", error: " << exc.what();
+    }
+  }
+  FileLock(const FileLock&) = delete;
+  FileLock(const FileLock&&) = delete;
+  FileLock& operator=(const FileLock&) = delete;
+  FileLock& operator=(const FileLock&&) = delete;
+
+ private:
+  const char* const path_{nullptr};
+  boost::filesystem::ofstream file_;
```

Comment:
Is this needed so that the file gets automagically created before instantiate the boost::file_lock? Seems like you could do that all in the constructor and not need to keep this open fd around?

Reply:
It has to be created before `lock_` is initialized in the initialization list.